### PR TITLE
v2.0

### PR DESCRIPTION
**Pull request by jonschlinkert**

### CODE REVIEW

**Comment by paulmillr** [comment]
Huge effort. Thanks @jonschlinkert!

**Comment by jonschlinkert** [comment]
@paulmillr thank you! Means a lot!

File: .verb.md [hunk 1]
```diff
@@ -88,14 +89,33 @@ console.log(isMatch('a/b.js')); //=> false
...
 | `posix`          | `boolean`      | `false`     | Support POSX character classes ("posix brackets"). |
+| `posixSlashes`   | `boolean`      | `undefined` | Convert all slashes in file paths to forward slashes. This does not convert slashes in the glob pattern itself |
 | `prepend`        | `boolean`      | `undefined` | String to prepend to the generated regex used for matching. |
 | `strictBrackets` | `boolean`      | `undefined` | Throw an error if brackets, braces, or parens are imbalanced. |
 | `strictSlashes`  | `boolean`      | `undefined` | Strictly enforce leading and trailing slashes. |
-| `unescapeRegex`  | `boolean`      | `undefined` | Remove backslashes preceding escaped characters in the returned regular expression. By default, backslashes are retained. |
-| `unixify`        | `boolean`      | `undefined` | Convert all slashes in the list to match (not in the glob pattern itself) to forward slashes. |
+| `unescape`       | `boolean`      | `undefined` | Remove backslashes preceding escaped characters in the generated regular expression. By default, backslashes are retained. |
 
+### options.onMatched
```
Comment (doowb) [inline .verb.md]:
```suggestion
### options.onMatch
```

File: .verb.md [hunk 2]
```diff
@@ -232,6 +252,24 @@ console.log(require('picomatch').makeRe('**/*').test('foo/bar/baz/qux.js'));
...
+
+**Accuracy**
+
+The number one of goal of this libary is accuracy. However, it's not unusual for different glob implementations to have different rules for matching behavior, even with simple wildcard matching. It gets increasingly more complicated when combinations of different features are combined, like when extglobs are combined with globstars, braces, slashes, and so on: `!(**/{a,b,*/c})`.
+
+Thus, given that there is no cannonical glob specification to use as a single source of truth when differences of opinion arise regarding behavior, sometimes we have to implement our best judgement and rely on feedback from users to make improvements. 
+
+**Performance**
+
+Although this library performs well in benchmarks, and in most cases it's faster than other popular libraries we benchmarked agains, we will always choose accuracy over performance. It's not helpful to anyone if our library is faster at returning the wrong answer.
```
Comment (doowb) [inline .verb.md]:
```suggestion
Although this library performs well in benchmarks, and in most cases it's faster than other popular libraries we benchmarked against, we will always choose accuracy over performance. It's not helpful to anyone if our library is faster at returning the wrong answer.
```

**Comment by jonschlinkert** [comment]
Windows tests are failing, but will be trivial to fix.

**Comment by paulmillr** [comment]
Do you think we’re able to do the release any time soon?

We’ve just merged an updated internal watcher for chokidar 3. Including new matcher would help a lot! Since the release is coming soon.

**Comment by jonschlinkert** [comment]
> We’ve just merged an updated internal watcher for chokidar 3. Including new matcher would help a lot! Since the release is coming soon.

Yes, absolutely, I'm just trying to fix a few windows tests. Almost there

**Comment by paulmillr** [comment]
@jonschlinkert ping

**Comment by doowb** [comment]
@paulmillr fwiw, @jonschlinkert  and I have been spending hours together each day on screen shares working on `picomatch/micromatch` and to ensure tests and edge cases are covered.

It's turning out to be awesome and I'm excited for the new version, but we just want to make sure we've covered the issues that people have brought up.

**Comment by jonschlinkert** [comment]
@paulmillr I generate documentation. Can we hold off on other nitpicks until I get this released? I'd like to keep merge conflicts to a minimum.

**Comment by paulmillr** [comment]
@jonschlinkert well, this nitpick made Type Errors in typescriptyish js. Just want to ensure this gets released with proper typings.

**Comment by jonschlinkert** [comment]
> this nitpick made Type Errors in typescriptyish js. Just want to ensure this gets released with proper typings

K, fair enough. Just wanted to mention it. I'm working on it now

**Comment by SimenB** [comment]
Gratz! 🎉
